added lucid item usability

kaincenteno opened the pull request:
changed casting time to 1 second and added script, now caskets should be livelier

1 thing missing would be lucid_wings_i and lucid_wings_ii that are still missing. If someone sees this PR and can mimic the same for those 2 items would be awesome =)

Thank you @Omnione  for making caskets available =)

<!-- place 'x' mark between square [] brackets to affirm: -->
**_I affirm:_**
- [x] that I agree to Project Topaz's [Limited Contributor License Agreement](http://project-topaz.com/blob/release/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I've _tested my code_ since the last commit in the PR, and will test after any later commits

kaincenteno committed:
added lucid item usability
changed casting time to 1 second and added script, now caskets should be livelier

1 thing missing would be lucid_wings_i and lucid_wings_ii that are still missing. If someone sees this PR and can mimic the same for those 2 items would be awesome =)

Thank you @mrsent for making caskets available =)

megberry commented on scripts/globals/items/lucid_elixir_i.lua:
```diff
@@ -0,0 +1,20 @@
+    target:addHP((target:getMaxHP()/100)*50*ITEM_POWER)
+    target:addMP((target:getMaxMP()/100)*50*ITEM_POWER)
+    target:messageBasic(tpz.msg.basic.RECOVERS_HP_AND_MP)
+end
```
add line feed to final lines, to match project standard

kaincenteno replied:
was not aware, ty <3

wrenffxi commented on scripts/globals/items/lucid_potion_iii.lua:
```diff
@@ -0,0 +1,18 @@
+end
+
+function onItemUse(target)
+    target:messageBasic(tpz.msg.basic.RECOVERS_HP, 0, target:addHP(1000*ITEM_POWER))
```
restores 2000 HP

kaincenteno replied:
thank you for review <3

wrenffxi commented on sql/item_usable.sql:
```diff
@@ -1623,14 +1623,14 @@ INSERT INTO `item_usable` VALUES (5820,'darkling_bolt_quiver',1,4,55,0,0,0,0,0);
+INSERT INTO `item_usable` VALUES (5824,'lucid_potion_i',1,1,30,0,0,0,0,0);
+INSERT INTO `item_usable` VALUES (5825,'lucid_potion_ii',1,1,31,0,0,0,0,0);
+INSERT INTO `item_usable` VALUES (5826,'lucid_potion_iii',1,1,31,0,0,0,0,0);
+INSERT INTO `item_usable` VALUES (5827,'lucid_ether_i',1,1,0,32,0,0,0,0);
```
the three inserts for ethers are missing animation (probably the value in animation time is supposed to be in the animation column)

kaincenteno replied:
omg ty i didnt notice <3

kaincenteno committed:
added lucid item usability
changed casting time to 1 second and added script, now caskets should be livelier

1 thing missing would be lucid_wings_i and lucid_wings_ii that are still missing. If someone sees this PR and can mimic the same for those 2 items would be awesome =)

Thank you @mrsent for making caskets available =)

kaincenteno commented:
Thank you for your review <3 I have made the changes asked, also followed teo's advise to reduce one division.

TeoTwawki commented on scripts/globals/items/lucid_elixir_ii.lua:
```diff
@@ -0,0 +1,20 @@
+end
+
+function onItemUse(target)
+    target:addHP(target:getMaxHP()*0.75*ITEM_POWER)
```
this is one of those case where "extra" parentheses can make the order of operations clearer.
(new people unfamiliar with C++ may wonder if the if item power is applied to the whole amount or to the .75 but this comment is not intended to say you need to change it, so long as it works correctly)

wrenffxi reviewed (approved):
Tried all eight items in-game.  All work as expected.
